From a129087bfd93f10c0a29c3903eaf0560bffadf17 Mon Sep 17 00:00:00 2001
From: Santhosh Kumar <santhosh.kumar.selvaraj@qt.io>
Date: Tue, 16 Sep 2025 10:22:26 +0200
Subject: [PATCH 043/239] Make security header for the qt labs folderlistmodel

The FolderListModel operates with QFileInfo and QUrl interfaces to
provide file information in the provided path. The path used needs to be
validated by the application developer for any illegitimate access.

QUIP: 23
Task-number: QTBUG-136184
Pick-to: 6.9 6.8
Change-Id: I50a4737aa9193a5b9996a37cd865c49de7b84121
Reviewed-by: Oliver Eftevaag <oliver.eftevaag@qt.io>
(cherry picked from commit 9ba8506e1b9eb15d17f9694f31e92bd12f134881)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/labs/folderlistmodel/fileinfothread.cpp        | 1 +
 src/labs/folderlistmodel/qquickfolderlistmodel.cpp | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/labs/folderlistmodel/fileinfothread.cpp b/src/labs/folderlistmodel/fileinfothread.cpp
index aa670338de..3faae0618f 100644
--- a/src/labs/folderlistmodel/fileinfothread.cpp
+++ b/src/labs/folderlistmodel/fileinfothread.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "fileinfothread_p.h"
 #include <qdiriterator.h>
diff --git a/src/labs/folderlistmodel/qquickfolderlistmodel.cpp b/src/labs/folderlistmodel/qquickfolderlistmodel.cpp
index 4f7f6d546f..a1f8a09778 100644
--- a/src/labs/folderlistmodel/qquickfolderlistmodel.cpp
+++ b/src/labs/folderlistmodel/qquickfolderlistmodel.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 //![code]
 #include "qquickfolderlistmodel_p.h"
-- 
2.51.2

